six/common/builtins/_util.c

```diff
@@ -91,84 +92,80 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
     char *exception = NULL;
     PyObject *cmd_args = NULL;
     PyObject *cmd_raise_on_empty = NULL;
-    PyObject *py_result = NULL;
+    PyObject *pyResult = NULL;
 
     PyGILState_STATE gstate = PyGILState_Ensure();
 
     if (!PyArg_ParseTuple(args, "O|O:get_subprocess_output", &cmd_args, &cmd_raise_on_empty)) {
-        goto error;
+        goto cleanup;
     }
 
     if (!PyList_Check(cmd_args)) {
         PyErr_SetString(PyExc_TypeError, "command args not a list");
-        goto error;
+        goto cleanup;
     }
 
+    // We already PyList_Check cmd_args, so PyList_Size won't fail and return -1
     subprocess_args_sz = PyList_Size(cmd_args);
     if (subprocess_args_sz == 0) {
         PyErr_SetString(PyExc_TypeError, "invalid command: empty list");
-        goto error;
+        goto cleanup;
     }
 
     if (!(subprocess_args = (char **)malloc(sizeof(*subprocess_args) * (subprocess_args_sz + 1)))) {
         PyErr_SetString(PyExc_MemoryError, "unable to allocate memory, bailing out");
-        goto error;
+        goto cleanup;
+    }
+
+    // init to NULL for safety - could use memset, but this is safer.
+    for (i = 0; i <= subprocess_args_sz; i++) {
+        subprocess_args[i] = NULL;
     }
 
-    subprocess_args[subprocess_args_sz] = NULL;
-    int i;
     for (i = 0; i < subprocess_args_sz; i++) {
         char *subprocess_arg = as_string(PyList_GetItem(cmd_args, i));
 
         if (subprocess_arg == NULL) {
             // cleanup
```

Comment:
Nit: I guess we can also remove the comment since the cleanup isn't done here anymore?

Reply:
```suggestion
```